pkg/api/v1/attribute_list_params.go

```diff
@@ -106,23 +137,35 @@ func (p *AttributeListParams) queryMods(tblName string) qm.QueryMod {
 		// injection since these strings are passed in by the user.
 		jsonPath += "?"
 
-		sqlValues = append(sqlValues, k)
+		values = append(values, k)
+	}
+
+	where, values := p.setJSONBWhereClause(tblName, jsonPath, values)
+
+	queryMods := []qm.QueryMod{nsMod, qm.And(where, values...)}
+
+	if p.AttributeOperator == AttributeLogicalOR {
+		return qm.Or2(qm.Expr(queryMods...))
 	}
```

Comment:
Is this logic correct? The way I read it, you're adding the `qm.And()` to your queryMods and OR'ing it afterwards. Shouldn't this be something like:
```
if p.AttributeOperator == AttributeLogicalOR {
		queryMods := []qm.QueryMod{nsMod, qm.Or2(where, values...)}
} else {
                 queryMods := []qm.QueryMod{nsMod, qm.And(where, values...)}
}

return qm.Expr(queryMods...)
```

Reply:
Ah, looked up the docs for `qm.Or2()` and it actually changes the where clause to an `OR`. Well, I guess it would be more straightforward to read if we used `qm.Or()` instead but it's up to you.

Reply:
Yeah, I'll [added](de8569c62dc0d1c53344857815e1b114f4ff4ee9) some comments so its a bit more clearer.

The `AND` in this [line](https://github.com/joelrebel/hollow-serverservice/blob/8f98b16cdfcd92c5a287271bdd53895d73c0b1f1/pkg/api/v1/attribute_list_params.go#L145) is for fields in the attribute itself[0]. 

We want an OR[1] between the two attributes, and because the *where* clause we have is only for the JSONB lookup, it cannot be passed to `qm.Or()`.

```sql
   WHERE
   (
      component_firmware_set_attr_0.namespace = 'sh.hollow.firmware_set.labels'
      AND -- *0
      json_extract_path_text(component_firmware_set_attr_0.data::JSONB, 'model') = 'r6515'
    )
    OR -- *1
    (
       component_firmware_set_attr_1.namespace = 'sh.hollow.firmware_set.labels'
        AND
       json_extract_path_text(component_firmware_set_attr_1.data::JSONB, 'model') = 'r640'
     );
```